app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -40,7 +40,11 @@ def check_user_enabled
   end
 
   def authorized
-    User.current.allowed_to?(path_to_authenticate)
+    if @available_sso.is_a?(SSO::Jwt) && @available_sso.jwt_scope&.any?
+      @available_sso.allowed_to?(path_to_authenticate)
```

Comment:
here we may need to also add `&& User.current.allowed_to?(path_to_authenticate)`, user could have lost permissions meanwhile, the token would still allow accessing these actions